src/commands/use.rs

```diff
@@ -19,6 +19,29 @@ pub struct Use {
     install_if_missing: bool,
 }
 
+impl Use {
+    fn install_new_version(
+        requested_version: UserVersion,
+        config: &FnmConfig,
+        install_if_missing: bool,
+    ) -> Result<(), Error> {
+        Install {
+            version: Some(requested_version.clone()),
+            ..Default::default()
+        }
+        .apply(config)
+        .context(InstallError)?;
+
+        Self {
+            version: Some(UserVersionReader::Direct(requested_version)),
+            install_if_missing,
+        }
+        .apply(config)?;
+
+        return Ok(());
+    }
+}
```

Comment:
can we move this out of the `impl Use`?

Dangling functions are okay -- `fn install_new_version` is all good. We can also move the `ensure` block inside to share it among the two branches.